<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Prepare the gene annotation databases — PrepareDB • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare the gene annotation databases — PrepareDB"><meta name="description" content="This function prepares the gene annotation databases for a given species and set of annotation sources.
It retrieves the necessary information from various annotation packages or external resources and organizes it into a list.
The list contains the annotation data for each specified annotation source."><meta property="og:description" content="This function prepares the gene annotation databases for a given species and set of annotation sources.
It retrieves the necessary information from various annotation packages or external resources and organizes it into a list.
The list contains the annotation data for each specified annotation source."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Prepare the gene annotation databases</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/main/R/SCP-analysis.R" class="external-link"><code>R/SCP-analysis.R</code></a></small>
      <div class="d-none name"><code>PrepareDB.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function prepares the gene annotation databases for a given species and set of annotation sources.
It retrieves the necessary information from various annotation packages or external resources and organizes it into a list.
The list contains the annotation data for each specified annotation source.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PrepareDB</span><span class="op">(</span></span>
<span>  species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo_sapiens"</span>, <span class="st">"Mus_musculus"</span><span class="op">)</span>,</span>
<span>  db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO"</span>, <span class="st">"GO_BP"</span>, <span class="st">"GO_CC"</span>, <span class="st">"GO_MF"</span>, <span class="st">"KEGG"</span>, <span class="st">"WikiPathway"</span>, <span class="st">"Reactome"</span>, <span class="st">"CORUM"</span>,</span>
<span>    <span class="st">"MP"</span>, <span class="st">"DO"</span>, <span class="st">"HPO"</span>, <span class="st">"PFAM"</span>, <span class="st">"CSPA"</span>, <span class="st">"Surfaceome"</span>, <span class="st">"SPRomeDB"</span>, <span class="st">"VerSeDa"</span>, <span class="st">"TFLink"</span>,</span>
<span>    <span class="st">"hTFtarget"</span>, <span class="st">"TRRUST"</span>, <span class="st">"JASPAR"</span>, <span class="st">"ENCODE"</span>, <span class="st">"MSigDB"</span>, <span class="st">"CellTalk"</span>, <span class="st">"CellChat"</span>,</span>
<span>    <span class="st">"Chromosome"</span>, <span class="st">"GeneType"</span>, <span class="st">"Enzyme"</span>, <span class="st">"TF"</span><span class="op">)</span>,</span>
<span>  db_IDtypes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"symbol"</span>, <span class="st">"entrez_id"</span>, <span class="st">"ensembl_id"</span><span class="op">)</span>,</span>
<span>  db_version <span class="op">=</span> <span class="st">"latest"</span>,</span>
<span>  db_update <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  convert_species <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Ensembl_version <span class="op">=</span> <span class="fl">103</span>,</span>
<span>  mirror <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  biomart <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  max_tries <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  custom_TERM2GENE <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_TERM2NAME <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_species <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_IDtype <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  custom_version <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-species">species<a class="anchor" aria-label="anchor" href="#arg-species"></a></dt>
<dd><p>A character vector specifying the species for which the gene annotation databases should be prepared. Default is c("Homo_sapiens", "Mus_musculus").</p></dd>


<dt id="arg-db">db<a class="anchor" aria-label="anchor" href="#arg-db"></a></dt>
<dd><p>A character vector specifying the annotation sources to be included in the gene annotation databases.
Default is c("GO", "GO_BP", "GO_CC", "GO_MF", "KEGG", "WikiPathway", "Reactome",
"CORUM", "MP", "DO", "HPO", "PFAM", "CSPA", "Surfaceome", "SPRomeDB", "VerSeDa",
"TFLink", "hTFtarget", "TRRUST", "JASPAR", "ENCODE", "MSigDB",
"CellTalk", "CellChat", "Chromosome", "GeneType", "Enzyme", "TF").</p></dd>


<dt id="arg-db-idtypes">db_IDtypes<a class="anchor" aria-label="anchor" href="#arg-db-idtypes"></a></dt>
<dd><p>A character vector specifying the desired ID types to be used for gene identifiers in the gene annotation databases. Default is c("symbol", "entrez_id", "ensembl_id").</p></dd>


<dt id="arg-db-version">db_version<a class="anchor" aria-label="anchor" href="#arg-db-version"></a></dt>
<dd><p>A character vector specifying the version of the gene annotation databases to be retrieved. Default is "latest".</p></dd>


<dt id="arg-db-update">db_update<a class="anchor" aria-label="anchor" href="#arg-db-update"></a></dt>
<dd><p>A logical value indicating whether the gene annotation databases should be forcefully updated. If set to FALSE, the function will attempt to load the cached databases instead. Default is FALSE.</p></dd>


<dt id="arg-convert-species">convert_species<a class="anchor" aria-label="anchor" href="#arg-convert-species"></a></dt>
<dd><p>A logical value indicating whether to use a species-converted database when the annotation is missing for the specified species. The default value is TRUE.</p></dd>


<dt id="arg-ensembl-version">Ensembl_version<a class="anchor" aria-label="anchor" href="#arg-ensembl-version"></a></dt>
<dd><p>Ensembl database version. If NULL, use the current release version.</p></dd>


<dt id="arg-mirror">mirror<a class="anchor" aria-label="anchor" href="#arg-mirror"></a></dt>
<dd><p>Specify an Ensembl mirror to connect to. The valid options here are 'www', 'uswest', 'useast', 'asia'.</p></dd>


<dt id="arg-biomart">biomart<a class="anchor" aria-label="anchor" href="#arg-biomart"></a></dt>
<dd><p>The name of the BioMart database that you want to connect to. Possible options include "ensembl", "protists_mart", "fungi_mart", and "plants_mart".</p></dd>


<dt id="arg-max-tries">max_tries<a class="anchor" aria-label="anchor" href="#arg-max-tries"></a></dt>
<dd><p>The maximum number of attempts to connect with the BioMart service.</p></dd>


<dt id="arg-custom-term-gene">custom_TERM2GENE<a class="anchor" aria-label="anchor" href="#arg-custom-term-gene"></a></dt>
<dd><p>A data frame containing a custom TERM2GENE mapping for the specified species and annotation source. Default is NULL.</p></dd>


<dt id="arg-custom-term-name">custom_TERM2NAME<a class="anchor" aria-label="anchor" href="#arg-custom-term-name"></a></dt>
<dd><p>A data frame containing a custom TERM2NAME mapping for the specified species and annotation source. Default is NULL.</p></dd>


<dt id="arg-custom-species">custom_species<a class="anchor" aria-label="anchor" href="#arg-custom-species"></a></dt>
<dd><p>A character vector specifying the species name to be used in a custom database. Default is NULL.</p></dd>


<dt id="arg-custom-idtype">custom_IDtype<a class="anchor" aria-label="anchor" href="#arg-custom-idtype"></a></dt>
<dd><p>A character vector specifying the ID type to be used in a custom database. Default is NULL.</p></dd>


<dt id="arg-custom-version">custom_version<a class="anchor" aria-label="anchor" href="#arg-custom-version"></a></dt>
<dd><p>A character vector specifying the version to be used in a custom database. Default is NULL.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list containing the prepared gene annotation databases:</p><ul><li><p><code>TERM2GENE:</code> mapping of gene identifiers to terms</p></li>
<li><p><code>TERM2NAME:</code> mapping of terms to their names</p></li>
<li><p><code>semData:</code> semantic similarity data for gene sets (only for Gene Ontology terms)</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The `PrepareDB` function prepares gene annotation databases for a given species and set of annotation sources.
It retrieves the necessary information from various annotation packages or external resources and organizes it into a list.
The function also supports creating custom databases based on user-provided gene sets.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="ListDB.html">ListDB</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="ListDB.html">ListDB</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Homo_sapiens"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO_BP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># Based on homologous gene conversion, prepare a gene annotation database that originally does not exist in the species.</span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>, db <span class="op">=</span> <span class="st">"MP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="ListDB.html">ListDB</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>, db <span class="op">=</span> <span class="st">"MP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Homo_sapiens"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"MP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># Prepare databases for other species</span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Macaca_fascicularis"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="ListDB.html">ListDB</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Macaca_fascicularis"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Macaca_fascicularis"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO_BP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Saccharomyces_cerevisiae"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="ListDB.html">ListDB</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Saccharomyces_cerevisiae"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Saccharomyces_cerevisiae"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"GO_BP"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># Prepare databases for Arabidopsis (plant)</span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span></span></span>
<span class="r-in"><span>    species <span class="op">=</span> <span class="st">"Arabidopsis_thaliana"</span>,</span></span>
<span class="r-in"><span>    db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>      <span class="st">"GO_BP"</span>, <span class="st">"GO_CC"</span>, <span class="st">"GO_MF"</span>, <span class="st">"KEGG"</span>, <span class="st">"WikiPathway"</span>,</span></span>
<span class="r-in"><span>      <span class="st">"ENZYME"</span>, <span class="st">"Chromosome"</span></span></span>
<span class="r-in"><span>    <span class="op">)</span>,</span></span>
<span class="r-in"><span>    biomart <span class="op">=</span> <span class="st">"plants_mart"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Arabidopsis_thaliana"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"KEGG"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># You can also build a custom database based on the gene sets you have</span></span></span>
<span class="r-in"><span>  <span class="va">ccgenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="CC_GenePrefetch.html">CC_GenePrefetch</a></span><span class="op">(</span><span class="st">"Homo_sapiens"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">custom_TERM2GENE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>term <span class="op">=</span> <span class="st">"S_genes"</span>, gene <span class="op">=</span> <span class="va">ccgenes</span><span class="op">[[</span><span class="st">"cc_S_genes"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>term <span class="op">=</span> <span class="st">"G2M_genes"</span>, gene <span class="op">=</span> <span class="va">ccgenes</span><span class="op">[[</span><span class="st">"cc_G2M_genes"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">custom_TERM2GENE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="co"># Set convert_species = TRUE to build a custom database for both species, with the name "CellCycle"</span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span></span></span>
<span class="r-in"><span>    species <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Homo_sapiens"</span>, <span class="st">"Mus_musculus"</span><span class="op">)</span>, db <span class="op">=</span> <span class="st">"CellCycle"</span>, convert_species <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    custom_TERM2GENE <span class="op">=</span> <span class="va">custom_TERM2GENE</span>, custom_species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>, custom_IDtype <span class="op">=</span> <span class="st">"symbol"</span>, custom_version <span class="op">=</span> <span class="st">"Seurat_v4"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="ListDB.html">ListDB</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"CellCycle"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>  <span class="va">db_list</span> <span class="op">&lt;-</span> <span class="fu">PrepareDB</span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus_musculus"</span>, db <span class="op">=</span> <span class="st">"CellCycle"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">db_list</span><span class="op">[[</span><span class="st">"Mus_musculus"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"CellCycle"</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="st">"TERM2GENE"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

