<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Run cell-level quality control for single cell RNA-seq data. — RunCellQC • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run cell-level quality control for single cell RNA-seq data. — RunCellQC"><meta name="description" content="This function handles multiple quality control methods for single-cell RNA-seq data."><meta property="og:description" content="This function handles multiple quality control methods for single-cell RNA-seq data."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Run cell-level quality control for single cell RNA-seq data.</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/main/R/SCP-cellqc.R" class="external-link"><code>R/SCP-cellqc.R</code></a></small>
      <div class="d-none name"><code>RunCellQC.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function handles multiple quality control methods for single-cell RNA-seq data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">RunCellQC</span><span class="op">(</span></span>
<span>  <span class="va">srt</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  split.by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  return_filtered <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  qc_metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"doublets"</span>, <span class="st">"outlier"</span>, <span class="st">"umi"</span>, <span class="st">"gene"</span>, <span class="st">"mito"</span>, <span class="st">"ribo"</span>, <span class="st">"ribo_mito_ratio"</span>,</span>
<span>    <span class="st">"species"</span><span class="op">)</span>,</span>
<span>  db_method <span class="op">=</span> <span class="st">"scDblFinder"</span>,</span>
<span>  db_rate <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  db_coefficient <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  outlier_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"log10_nCount:lower:2.5"</span>, <span class="st">"log10_nCount:higher:5"</span>,</span>
<span>    <span class="st">"log10_nFeature:lower:2.5"</span>, <span class="st">"log10_nFeature:higher:5"</span>, <span class="st">"featurecount_dist:lower:2.5"</span><span class="op">)</span>,</span>
<span>  outlier_n <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  UMI_threshold <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  gene_threshold <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  mito_threshold <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  mito_pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MT-"</span>, <span class="st">"Mt-"</span>, <span class="st">"mt-"</span><span class="op">)</span>,</span>
<span>  mito_gene <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ribo_threshold <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  ribo_pattern <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RP[SL]\\d+\\w{0,1}\\d*$"</span>, <span class="st">"Rp[sl]\\d+\\w{0,1}\\d*$"</span>,</span>
<span>    <span class="st">"rp[sl]\\d+\\w{0,1}\\d*$"</span><span class="op">)</span>,</span>
<span>  ribo_gene <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ribo_mito_ratio_range <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="cn">Inf</span><span class="op">)</span>,</span>
<span>  species <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  species_gene_prefix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  species_percent <span class="op">=</span> <span class="fl">95</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">11</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-srt">srt<a class="anchor" aria-label="anchor" href="#arg-srt"></a></dt>
<dd><p>A Seurat object.</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>The name of the assay to be used for doublet-calling. Default is "RNA".</p></dd>


<dt id="arg-split-by">split.by<a class="anchor" aria-label="anchor" href="#arg-split-by"></a></dt>
<dd><p>Name of the sample variable to split the Seurat object. Default is NULL.</p></dd>


<dt id="arg-return-filtered">return_filtered<a class="anchor" aria-label="anchor" href="#arg-return-filtered"></a></dt>
<dd><p>Logical indicating whether to return a cell-filtered Seurat object. Default is FALSE.</p></dd>


<dt id="arg-qc-metrics">qc_metrics<a class="anchor" aria-label="anchor" href="#arg-qc-metrics"></a></dt>
<dd><p>A character vector specifying the quality control metrics to be applied. Default is
`c("doublets", "outlier", "umi", "gene", "mito", "ribo", "ribo_mito_ratio", "species")`.</p></dd>


<dt id="arg-db-method">db_method<a class="anchor" aria-label="anchor" href="#arg-db-method"></a></dt>
<dd><p>Doublet-calling methods used. Can be one of <code>scDblFinder</code>, <code>Scrublet</code>, <code>DoubletDetection</code>, <code>scds_cxds</code>, <code>scds_bcds</code>, <code>scds_hybrid</code></p></dd>


<dt id="arg-db-rate">db_rate<a class="anchor" aria-label="anchor" href="#arg-db-rate"></a></dt>
<dd><p>The expected doublet rate. By default this is assumed to be 1% per thousand cells captured (so 4% among 4000 thousand cells), which is appropriate for 10x datasets.</p></dd>


<dt id="arg-db-coefficient">db_coefficient<a class="anchor" aria-label="anchor" href="#arg-db-coefficient"></a></dt>
<dd><p>The coefficient used to calculate the doublet rate. Default is 0.01. Doublet rate is calculated as`ncol(srt) / 1000 * db_coefficient`</p></dd>


<dt id="arg-outlier-threshold">outlier_threshold<a class="anchor" aria-label="anchor" href="#arg-outlier-threshold"></a></dt>
<dd><p>A character vector specifying the outlier threshold. Default is
`c("log10_nCount:lower:2.5", "log10_nCount:higher:5", "log10_nFeature:lower:2.5", "log10_nFeature:higher:5", "featurecount_dist:lower:2.5")`. See <a href="https://rdrr.io/pkg/scuttle/man/isOutlier.html" class="external-link">isOutlier</a>.</p></dd>


<dt id="arg-outlier-n">outlier_n<a class="anchor" aria-label="anchor" href="#arg-outlier-n"></a></dt>
<dd><p>Minimum number of outlier metrics that meet the conditions for determining outlier cells. Default is 1.</p></dd>


<dt id="arg-umi-threshold">UMI_threshold<a class="anchor" aria-label="anchor" href="#arg-umi-threshold"></a></dt>
<dd><p>UMI number threshold. Cells that exceed this threshold will be considered as kept. Default is 3000.</p></dd>


<dt id="arg-gene-threshold">gene_threshold<a class="anchor" aria-label="anchor" href="#arg-gene-threshold"></a></dt>
<dd><p>Gene number threshold. Cells that exceed this threshold will be considered as kept. Default is 1000.</p></dd>


<dt id="arg-mito-threshold">mito_threshold<a class="anchor" aria-label="anchor" href="#arg-mito-threshold"></a></dt>
<dd><p>Percentage of UMI counts of mitochondrial genes. Cells that exceed this threshold will be considered as discarded. Default is 20.</p></dd>


<dt id="arg-mito-pattern">mito_pattern<a class="anchor" aria-label="anchor" href="#arg-mito-pattern"></a></dt>
<dd><p>Regex patterns to match the mitochondrial genes. Default is `c("MT-", "Mt-", "mt-")`.</p></dd>


<dt id="arg-mito-gene">mito_gene<a class="anchor" aria-label="anchor" href="#arg-mito-gene"></a></dt>
<dd><p>A defined mitochondrial genes. If features provided, will ignore the <code>mito_pattern</code> matching. Default is <code>NULL</code>.</p></dd>


<dt id="arg-ribo-threshold">ribo_threshold<a class="anchor" aria-label="anchor" href="#arg-ribo-threshold"></a></dt>
<dd><p>Percentage of UMI counts of ribosomal genes. Cells that exceed this threshold will be considered as discarded. Default is 50.</p></dd>


<dt id="arg-ribo-pattern">ribo_pattern<a class="anchor" aria-label="anchor" href="#arg-ribo-pattern"></a></dt>
<dd><p>Regex patterns to match the ribosomal genes. Default is `c("RP[SL]\d+\w0,1\d*$", "Rp[sl]\d+\w0,1\d*$", "rp[sl]\d+\w0,1\d*$")`.</p></dd>


<dt id="arg-ribo-gene">ribo_gene<a class="anchor" aria-label="anchor" href="#arg-ribo-gene"></a></dt>
<dd><p>A defined ribosomal genes. If features provided, will ignore the <code>ribo_pattern</code> matching. Default is <code>NULL</code>.</p></dd>


<dt id="arg-ribo-mito-ratio-range">ribo_mito_ratio_range<a class="anchor" aria-label="anchor" href="#arg-ribo-mito-ratio-range"></a></dt>
<dd><p>A numeric vector specifying the range of ribosomal/mitochondrial gene expression ratios for ribo_mito_ratio outlier cells. Default is c(1, Inf).</p></dd>


<dt id="arg-species">species<a class="anchor" aria-label="anchor" href="#arg-species"></a></dt>
<dd><p>Species used as the suffix of the QC metrics. The first is the species of interest. Default is <code>NULL</code>.</p></dd>


<dt id="arg-species-gene-prefix">species_gene_prefix<a class="anchor" aria-label="anchor" href="#arg-species-gene-prefix"></a></dt>
<dd><p>Species gene prefix used to calculate QC metrics for each species. Default is <code>NULL</code>.</p></dd>


<dt id="arg-species-percent">species_percent<a class="anchor" aria-label="anchor" href="#arg-species-percent"></a></dt>
<dd><p>Percentage of UMI counts of the first species. Cells that exceed this threshold will be considered as kept. Default is 95.</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Set a random seed. Default is 11.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns Seurat object with the QC results stored in the meta.data layer.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreas_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunCellQC</span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="fu"><a href="CellStatPlot.html">CellStatPlot</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">pancreas_sub</span>,</span></span>
<span class="r-in"><span>  stat.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"db_qc"</span>, <span class="st">"outlier_qc"</span>, <span class="st">"umi_qc"</span>, <span class="st">"gene_qc"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"mito_qc"</span>, <span class="st">"ribo_qc"</span>, <span class="st">"ribo_mito_ratio_qc"</span>, <span class="st">"species_qc"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  plot_type <span class="op">=</span> <span class="st">"upset"</span>, stat_level <span class="op">=</span> <span class="st">"Fail"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in StatPlot(meta.data = meta.data, stat.by = stat.by, group.by = group.by,     split.by = split.by, bg.by = bg.by, flip = flip, NA_color = NA_color,     NA_stat = NA_stat, keep_empty = keep_empty, individual = individual,     stat_level = stat_level, plot_type = plot_type, stat_type = stat_type,     position = position, palette = palette, palcolor = palcolor,     alpha = alpha, bg_palette = bg_palette, bg_palcolor = bg_palcolor,     bg_alpha = bg_alpha, label = label, label.size = label.size,     label.fg = label.fg, label.bg = label.bg, label.bg.r = label.bg.r,     aspect.ratio = aspect.ratio, title = title, subtitle = subtitle,     xlab = xlab, ylab = ylab, legend.position = legend.position,     legend.direction = legend.direction, theme_use = theme_use,     theme_args = theme_args, combine = combine, nrow = nrow,     ncol = ncol, byrow = byrow, force = force, seed = seed):</span> db_qc is not in the meta.data.</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">$</span><span class="va">CellQC</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in x[[i, drop = TRUE]]:</span> ‘CellQC’ not found in this Seurat object</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>  Did you mean "CellType"?</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"ifnb_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ifnb_sub</span> <span class="op">&lt;-</span> <span class="fu">RunCellQC</span><span class="op">(</span><span class="va">ifnb_sub</span>, split.by <span class="op">=</span> <span class="st">"stim"</span>, UMI_threshold <span class="op">=</span> <span class="fl">1000</span>, gene_threshold <span class="op">=</span> <span class="fl">550</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="fu"><a href="CellStatPlot.html">CellStatPlot</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">ifnb_sub</span>,</span></span>
<span class="r-in"><span>  stat.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"db_qc"</span>, <span class="st">"outlier_qc"</span>, <span class="st">"umi_qc"</span>, <span class="st">"gene_qc"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"mito_qc"</span>, <span class="st">"ribo_qc"</span>, <span class="st">"ribo_mito_ratio_qc"</span>, <span class="st">"species_qc"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span>,</span></span>
<span class="r-in"><span>  plot_type <span class="op">=</span> <span class="st">"upset"</span>, stat_level <span class="op">=</span> <span class="st">"Fail"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in StatPlot(meta.data = meta.data, stat.by = stat.by, group.by = group.by,     split.by = split.by, bg.by = bg.by, flip = flip, NA_color = NA_color,     NA_stat = NA_stat, keep_empty = keep_empty, individual = individual,     stat_level = stat_level, plot_type = plot_type, stat_type = stat_type,     position = position, palette = palette, palcolor = palcolor,     alpha = alpha, bg_palette = bg_palette, bg_palcolor = bg_palcolor,     bg_alpha = bg_alpha, label = label, label.size = label.size,     label.fg = label.fg, label.bg = label.bg, label.bg.r = label.bg.r,     aspect.ratio = aspect.ratio, title = title, subtitle = subtitle,     xlab = xlab, ylab = ylab, legend.position = legend.position,     legend.direction = legend.direction, theme_use = theme_use,     theme_args = theme_args, combine = combine, nrow = nrow,     ncol = ncol, byrow = byrow, force = force, seed = seed):</span> db_qc is not in the meta.data.</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ifnb_sub</span><span class="op">$</span><span class="va">CellQC</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in x[[i, drop = TRUE]]:</span> ‘CellQC’ not found in this Seurat object</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>  </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

