<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Differential gene test — RunDEtest • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential gene test — RunDEtest"><meta name="description" content="This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis."><meta property="og:description" content="This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Differential gene test</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/main/R/SCP-analysis.R" class="external-link"><code>R/SCP-analysis.R</code></a></small>
      <div class="d-none name"><code>RunDEtest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">RunDEtest</span><span class="op">(</span></span>
<span>  <span class="va">srt</span>,</span>
<span>  group_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  features <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  markers_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"paired"</span>, <span class="st">"conserved"</span>, <span class="st">"disturbed"</span><span class="op">)</span>,</span>
<span>  grouping.var <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  meta.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"maximump"</span>, <span class="st">"minimump"</span>, <span class="st">"wilkinsonp"</span>, <span class="st">"meanp"</span>, <span class="st">"sump"</span>, <span class="st">"votep"</span><span class="op">)</span>,</span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fc.threshold <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  base <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  pseudocount.use <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  mean.fxn <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.pct <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  min.diff.pct <span class="op">=</span> <span class="op">-</span><span class="cn">Inf</span>,</span>
<span>  max.cells.per.ident <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  latent.vars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.cells.feature <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  min.cells.group <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  norm.method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>  p.adjust.method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">11</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-srt">srt<a class="anchor" aria-label="anchor" href="#arg-srt"></a></dt>
<dd><p>A Seurat object.</p></dd>


<dt id="arg-group-by">group_by<a class="anchor" aria-label="anchor" href="#arg-group-by"></a></dt>
<dd><p>A grouping variable in the dataset to define the groups or conditions for the differential test. If not provided, the function uses the "active.ident" variable in the Seurat object.</p></dd>


<dt id="arg-group-">group1<a class="anchor" aria-label="anchor" href="#arg-group-"></a></dt>
<dd><p>A vector of cell IDs or a character vector specifying the cells that belong to the first group. If both group_by and group1 are provided, group1 takes precedence.</p></dd>


<dt id="arg-group-">group2<a class="anchor" aria-label="anchor" href="#arg-group-"></a></dt>
<dd><p>A vector of cell IDs or a character vector specifying the cells that belong to the second group. This parameter is only used when group_by or group1 is provided.</p></dd>


<dt id="arg-cells-">cells1<a class="anchor" aria-label="anchor" href="#arg-cells-"></a></dt>
<dd><p>A vector of cell IDs specifying the cells that belong to group1. If provided, group1 is ignored.</p></dd>


<dt id="arg-cells-">cells2<a class="anchor" aria-label="anchor" href="#arg-cells-"></a></dt>
<dd><p>A vector of cell IDs specifying the cells that belong to group2. This parameter is only used when cells1 is provided.</p></dd>


<dt id="arg-features">features<a class="anchor" aria-label="anchor" href="#arg-features"></a></dt>
<dd><p>A vector of gene names specifying the features to consider for the differential test. If not provided, all features in the dataset are considered.</p></dd>


<dt id="arg-markers-type">markers_type<a class="anchor" aria-label="anchor" href="#arg-markers-type"></a></dt>
<dd><p>A character value specifying the type of markers to find. Possible values are "all", "paired", "conserved", and "disturbed".</p></dd>


<dt id="arg-grouping-var">grouping.var<a class="anchor" aria-label="anchor" href="#arg-grouping-var"></a></dt>
<dd><p>A character value specifying the grouping variable for finding conserved or disturbed markers. This parameter is only used when markers_type is "conserved" or "disturbed".</p></dd>


<dt id="arg-meta-method">meta.method<a class="anchor" aria-label="anchor" href="#arg-meta-method"></a></dt>
<dd><p>A character value specifying the method to use for combining p-values in the conserved markers test. Possible values are "maximump", "minimump", "wilkinsonp", "meanp", "sump", and "votep".</p></dd>


<dt id="arg-test-use">test.use<a class="anchor" aria-label="anchor" href="#arg-test-use"></a></dt>
<dd><p>Denotes which test to use. Available options are:</p><ul><li><p>"wilcox" : Identifies differentially expressed genes between two
 groups of cells using a Wilcoxon Rank Sum test (default); will use a fast
 implementation by Presto if installed</p></li>
<li><p>"wilcox_limma" : Identifies differentially expressed genes between two
 groups of cells using the limma implementation of the Wilcoxon Rank Sum test;
 set this option to reproduce results from Seurat v4</p></li>
<li><p>"bimod" : Likelihood-ratio test for single cell gene expression,
 (McDavid et al., Bioinformatics, 2013)</p></li>
<li><p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
 For each gene, evaluates (using AUC) a classifier built on that gene alone,
 to classify between two groups of cells. An AUC value of 1 means that
 expression values for this gene alone can perfectly classify the two
 groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
 each of the cells in cells.2). An AUC value of 0 also means there is perfect
 classification, but in the other direction. A value of 0.5 implies that
 the gene has no predictive power to classify the two groups. Returns a
 'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
 expressed genes.</p></li>
<li><p>"t" : Identify differentially expressed genes between two groups of
 cells using the Student's t-test.</p></li>
<li><p>"negbinom" : Identifies differentially expressed genes between two
  groups of cells using a negative binomial generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"poisson" : Identifies differentially expressed genes between two
  groups of cells using a poisson generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"LR" : Uses a logistic regression framework to determine differentially
 expressed genes. Constructs a logistic regression model predicting group
 membership based on each feature individually and compares this to a null
 model with a likelihood ratio test.</p></li>
<li><p>"MAST" : Identifies differentially expressed genes between two groups
 of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
 package to run the DE testing.</p></li>
<li><p>"DESeq2" : Identifies differentially expressed genes between two groups
 of cells based on a model using DESeq2 which uses a negative binomial
 distribution (Love et al, Genome Biology, 2014).This test does not support
 pre-filtering of genes based on average difference (or percent detection rate)
 between cell groups. However, genes may be pre-filtered based on their
 minimum detection rate (min.pct) across both cell groups. To use this method,
 please install DESeq2, using the instructions at
 https://bioconductor.org/packages/release/bioc/html/DESeq2.html</p></li>
</ul></dd>


<dt id="arg-only-pos">only.pos<a class="anchor" aria-label="anchor" href="#arg-only-pos"></a></dt>
<dd><p>Only return positive markers (FALSE by default)</p></dd>


<dt id="arg-fc-threshold">fc.threshold<a class="anchor" aria-label="anchor" href="#arg-fc-threshold"></a></dt>
<dd><p>A numeric value used to filter genes for testing based on their average fold change between/among the two groups. By default, it is set to 1.5</p></dd>


<dt id="arg-base">base<a class="anchor" aria-label="anchor" href="#arg-base"></a></dt>
<dd><p>The base with respect to which logarithms are computed.</p></dd>


<dt id="arg-pseudocount-use">pseudocount.use<a class="anchor" aria-label="anchor" href="#arg-pseudocount-use"></a></dt>
<dd><p>Pseudocount to add to averaged expression values when
calculating logFC. 1 by default.</p></dd>


<dt id="arg-mean-fxn">mean.fxn<a class="anchor" aria-label="anchor" href="#arg-mean-fxn"></a></dt>
<dd><p>Function to use for fold change or average difference calculation.
The default depends on the the value of <code>fc.slot</code>:</p><ul><li><p>"counts" : difference in the log of the mean counts, with pseudocount.</p></li>
<li><p>"data" : difference in the log of the average exponentiated data, with pseudocount.
 This adjusts for differences in sequencing depth between cells, and assumes that "data"
 has been log-normalized.</p></li>
<li><p>"scale.data" : difference in the means of scale.data.</p></li>
</ul></dd>


<dt id="arg-min-pct">min.pct<a class="anchor" aria-label="anchor" href="#arg-min-pct"></a></dt>
<dd><p>only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. Meant to speed up the function
by not testing genes that are very infrequently expressed. Default is 0.01</p></dd>


<dt id="arg-min-diff-pct">min.diff.pct<a class="anchor" aria-label="anchor" href="#arg-min-diff-pct"></a></dt>
<dd><p>only test genes that show a minimum difference in the
fraction of detection between the two groups. Set to -Inf by default</p></dd>


<dt id="arg-max-cells-per-ident">max.cells.per.ident<a class="anchor" aria-label="anchor" href="#arg-max-cells-per-ident"></a></dt>
<dd><p>Down sample each identity class to a max number.
Default is no downsampling. Not activated by default (set to Inf)</p></dd>


<dt id="arg-latent-vars">latent.vars<a class="anchor" aria-label="anchor" href="#arg-latent-vars"></a></dt>
<dd><p>Variables to test, used only when <code>test.use</code> is one of
'LR', 'negbinom', 'poisson', or 'MAST'</p></dd>


<dt id="arg-min-cells-feature">min.cells.feature<a class="anchor" aria-label="anchor" href="#arg-min-cells-feature"></a></dt>
<dd><p>Minimum number of cells expressing the feature in at least one
of the two groups, currently only used for poisson and negative binomial tests</p></dd>


<dt id="arg-min-cells-group">min.cells.group<a class="anchor" aria-label="anchor" href="#arg-min-cells-group"></a></dt>
<dd><p>Minimum number of cells in one of the groups</p></dd>


<dt id="arg-norm-method">norm.method<a class="anchor" aria-label="anchor" href="#arg-norm-method"></a></dt>
<dd><p>Normalization method for fold change calculation when slot is 'data'. Default is "LogNormalize".</p></dd>


<dt id="arg-p-adjust-method">p.adjust.method<a class="anchor" aria-label="anchor" href="#arg-p-adjust-method"></a></dt>
<dd><p>A character value specifying the method to use for adjusting p-values. Default is "bonferroni".</p></dd>


<dt id="arg-slot">slot<a class="anchor" aria-label="anchor" href="#arg-slot"></a></dt>
<dd><p>Slot to pull data from; note that if <code>test.use</code> is
"negbinom", "poisson", or "DESeq2", <code>slot</code> will be set to "counts"</p></dd>


<dt id="arg-assay">assay<a class="anchor" aria-label="anchor" href="#arg-assay"></a></dt>
<dd><p>Assay to use in differential expression testing</p></dd>


<dt id="arg-bpparam">BPPARAM<a class="anchor" aria-label="anchor" href="#arg-bpparam"></a></dt>
<dd><p>A BiocParallelParam object specifying the parallelization parameters for the differential test. Default is BiocParallel::bpparam().</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>An integer value specifying the seed. Default is 11.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>A logical value indicating whether to display progress messages during the differential test. Default is TRUE.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments to pass to the <code><a href="https://satijalab.org/seurat/reference/FindMarkers.html" class="external-link">FindMarkers</a></code> function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="RunEnrichment.html">RunEnrichment</a></code> <code><a href="RunGSEA.html">RunGSEA</a></code> <code><a href="GroupHeatmap.html">GroupHeatmap</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreas_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">AllMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_SubCellType</span><span class="op">$</span><span class="va">AllMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">AllMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'AllMarkers' not found</span>
<span class="r-in"><span><span class="va">ht1</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">AllMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">AllMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'AllMarkers' not found</span>
<span class="r-in"><span><span class="va">ht1</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht1' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">TopMarkers</span> <span class="op">&lt;-</span> <span class="va">AllMarkers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">avg_log2FC</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'AllMarkers' not found</span>
<span class="r-in"><span><span class="va">ht2</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">TopMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">TopMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span>, show_row_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'TopMarkers' not found</span>
<span class="r-in"><span><span class="va">ht2</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht2' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"SubCellType"</span>, markers_type <span class="op">=</span> <span class="st">"paired"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">PairedMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_SubCellType</span><span class="op">$</span><span class="va">PairedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">PairedMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'PairedMarkers' not found</span>
<span class="r-in"><span><span class="va">ht3</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">PairedMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">PairedMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'PairedMarkers' not found</span>
<span class="r-in"><span><span class="va">ht3</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht3' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"panc8_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="Integration_SCP.html">Integration_SCP</a></span><span class="op">(</span><span class="va">panc8_sub</span>, batch <span class="op">=</span> <span class="st">"tech"</span>, integration_method <span class="op">=</span> <span class="st">"Seurat"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2025-09-07 19:45:38.577593] Start Seurat_integrate</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in invoke(.fn = paste0(integration_method, "_integrate"), .args = args[names(args) %in%     formalArgs(paste0(integration_method, "_integrate"))]):</span> could not find function "invoke"</span>
<span class="r-in"><span><span class="fu"><a href="CellDimPlot.html">CellDimPlot</a></span><span class="op">(</span><span class="va">panc8_sub</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">"tech"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in DefaultReduction(srt):</span> Unable to find any reductions.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, markers_type <span class="op">=</span> <span class="st">"conserved"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">ConservedMarkers1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_celltype</span><span class="op">$</span><span class="va">ConservedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ConservedMarkers1' not found</span>
<span class="r-in"><span><span class="va">ht4</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"tech"</span>, split.by <span class="op">=</span> <span class="st">"celltype"</span>, within_groups <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ConservedMarkers1' not found</span>
<span class="r-in"><span><span class="va">ht4</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht4' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"tech"</span>, grouping.var <span class="op">=</span> <span class="st">"celltype"</span>, markers_type <span class="op">=</span> <span class="st">"conserved"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">ConservedMarkers2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_tech</span><span class="op">$</span><span class="va">ConservedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ConservedMarkers2' not found</span>
<span class="r-in"><span><span class="va">ht4</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"tech"</span>, split.by <span class="op">=</span> <span class="st">"celltype"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ConservedMarkers2' not found</span>
<span class="r-in"><span><span class="va">ht4</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht4' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, markers_type <span class="op">=</span> <span class="st">"disturbed"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">DisturbedMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_celltype</span><span class="op">$</span><span class="va">DisturbedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">var1</span> <span class="op">==</span> <span class="st">"smartseq2"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers' not found</span>
<span class="r-in"><span><span class="va">ht5</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, split.by <span class="op">=</span> <span class="st">"tech"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers' not found</span>
<span class="r-in"><span><span class="va">ht5</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht5' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gene_specific</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'which': object 'DisturbedMarkers' not found</span>
<span class="r-in"><span><span class="va">DisturbedMarkers_specific</span> <span class="op">&lt;-</span> <span class="va">DisturbedMarkers</span><span class="op">[</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gene_specific</span>, <span class="op">]</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers' not found</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers_specific' not found</span>
<span class="r-in"><span><span class="va">ht6</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, split.by <span class="op">=</span> <span class="st">"tech"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers_specific' not found</span>
<span class="r-in"><span><span class="va">ht6</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht6' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ht7</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>, aggregate_fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">expm1</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, numerator <span class="op">=</span> <span class="st">"smartseq2"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'DisturbedMarkers_specific' not found</span>
<span class="r-in"><span><span class="va">ht7</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'ht7' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

