<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Perform the enrichment analysis (GSEA) on the genes — RunGSEA • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Perform the enrichment analysis (GSEA) on the genes — RunGSEA"><meta name="description" content="Perform the enrichment analysis (GSEA) on the genes"><meta property="og:description" content="Perform the enrichment analysis (GSEA) on the genes"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.6.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Perform the enrichment analysis (GSEA) on the genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/main/R/SCP-analysis.R" class="external-link"><code>R/SCP-analysis.R</code></a></small>
      <div class="d-none name"><code>RunGSEA.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Perform the enrichment analysis (GSEA) on the genes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">RunGSEA</span><span class="op">(</span></span>
<span>  srt <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  DE_threshold <span class="op">=</span> <span class="st">"p_val_adj &lt; 0.05"</span>,</span>
<span>  scoreType <span class="op">=</span> <span class="st">"std"</span>,</span>
<span>  geneID <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneScore <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneID_groups <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneID_exclude <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  IDtype <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>  result_IDtype <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>,</span>
<span>  db <span class="op">=</span> <span class="st">"GO_BP"</span>,</span>
<span>  db_update <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  db_version <span class="op">=</span> <span class="st">"latest"</span>,</span>
<span>  db_combine <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  convert_species <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Ensembl_version <span class="op">=</span> <span class="fl">103</span>,</span>
<span>  mirror <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  TERM2GENE <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  TERM2NAME <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minGSSize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  maxGSSize <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  unlimited_db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="st">"GeneType"</span>, <span class="st">"TF"</span>, <span class="st">"Enzyme"</span>, <span class="st">"CSPA"</span><span class="op">)</span>,</span>
<span>  GO_simplify <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  GO_simplify_cutoff <span class="op">=</span> <span class="st">"p.adjust &lt; 0.05"</span>,</span>
<span>  simplify_method <span class="op">=</span> <span class="st">"Wang"</span>,</span>
<span>  simplify_similarityCutoff <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">11</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-srt">srt<a class="anchor" aria-label="anchor" href="#arg-srt"></a></dt>
<dd><p>A Seurat object containing the results of differential expression analysis (RunDEtest).
If specified, the genes and groups will be extracted from the Seurat object automatically.
If not specified, the <code>geneID</code> and <code>geneID_groups</code> arguments must be provided.</p></dd>


<dt id="arg-group-by">group_by<a class="anchor" aria-label="anchor" href="#arg-group-by"></a></dt>
<dd><p>A character vector specifying the grouping variable in the Seurat object. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt id="arg-test-use">test.use<a class="anchor" aria-label="anchor" href="#arg-test-use"></a></dt>
<dd><p>A character vector specifying the test to be used in differential expression analysis. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt id="arg-de-threshold">DE_threshold<a class="anchor" aria-label="anchor" href="#arg-de-threshold"></a></dt>
<dd><p>A character vector specifying the filter condition for differential expression analysis. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt id="arg-scoretype">scoreType<a class="anchor" aria-label="anchor" href="#arg-scoretype"></a></dt>
<dd><p>This parameter defines the GSEA score type. Possible options are "std", "pos", "neg". By default ("std") the enrichment score is computed as in the original GSEA. The "pos" and "neg" score types are intended to be used for one-tailed tests (i.e. when one is interested only in positive ("pos") or negateive ("neg") enrichment).</p></dd>


<dt id="arg-geneid">geneID<a class="anchor" aria-label="anchor" href="#arg-geneid"></a></dt>
<dd><p>A character vector specifying the gene IDs.</p></dd>


<dt id="arg-genescore">geneScore<a class="anchor" aria-label="anchor" href="#arg-genescore"></a></dt>
<dd><p>A numeric vector that specifies the gene scores, for example, the log2(fold change) values of gene expression.</p></dd>


<dt id="arg-geneid-groups">geneID_groups<a class="anchor" aria-label="anchor" href="#arg-geneid-groups"></a></dt>
<dd><p>A factor vector specifying the group labels for each gene.</p></dd>


<dt id="arg-geneid-exclude">geneID_exclude<a class="anchor" aria-label="anchor" href="#arg-geneid-exclude"></a></dt>
<dd><p>A character vector specifying the gene IDs to be excluded from the analysis.</p></dd>


<dt id="arg-idtype">IDtype<a class="anchor" aria-label="anchor" href="#arg-idtype"></a></dt>
<dd><p>A character vector specifying the type of gene IDs in the <code>srt</code> object or <code>geneID</code> argument. This argument is used to convert the gene IDs to a different type if <code>IDtype</code> is different from <code>result_IDtype</code>.</p></dd>


<dt id="arg-result-idtype">result_IDtype<a class="anchor" aria-label="anchor" href="#arg-result-idtype"></a></dt>
<dd><p>A character vector specifying the desired type of gene ID to be used in the output. This argument is used to convert the gene IDs from <code>IDtype</code> to <code>result_IDtype</code>.</p></dd>


<dt id="arg-species">species<a class="anchor" aria-label="anchor" href="#arg-species"></a></dt>
<dd><p>A character vector specifying the species for which the analysis is performed.</p></dd>


<dt id="arg-db">db<a class="anchor" aria-label="anchor" href="#arg-db"></a></dt>
<dd><p>A character vector specifying the name of the database to be used for enrichment analysis.</p></dd>


<dt id="arg-db-update">db_update<a class="anchor" aria-label="anchor" href="#arg-db-update"></a></dt>
<dd><p>A logical value indicating whether the gene annotation databases should be forcefully updated. If set to FALSE, the function will attempt to load the cached databases instead. Default is FALSE.</p></dd>


<dt id="arg-db-version">db_version<a class="anchor" aria-label="anchor" href="#arg-db-version"></a></dt>
<dd><p>A character vector specifying the version of the database to be used. This argument is ignored if <code>db_update</code> is <code>TRUE</code>. Default is "latest".</p></dd>


<dt id="arg-db-combine">db_combine<a class="anchor" aria-label="anchor" href="#arg-db-combine"></a></dt>
<dd><p>A logical value indicating whether to combine multiple databases into one. If TRUE, all database specified by <code>db</code> will be combined as one named "Combined".</p></dd>


<dt id="arg-convert-species">convert_species<a class="anchor" aria-label="anchor" href="#arg-convert-species"></a></dt>
<dd><p>A logical value indicating whether to use a species-converted database when the annotation is missing for the specified species. The default value is TRUE.</p></dd>


<dt id="arg-ensembl-version">Ensembl_version<a class="anchor" aria-label="anchor" href="#arg-ensembl-version"></a></dt>
<dd><p>Ensembl database version. If NULL, use the current release version.</p></dd>


<dt id="arg-mirror">mirror<a class="anchor" aria-label="anchor" href="#arg-mirror"></a></dt>
<dd><p>Specify an Ensembl mirror to connect to. The valid options here are 'www', 'uswest', 'useast', 'asia'.</p></dd>


<dt id="arg-term-gene">TERM2GENE<a class="anchor" aria-label="anchor" href="#arg-term-gene"></a></dt>
<dd><p>A data frame specifying the gene-term mapping for a custom database. The first column should contain the term IDs, and the second column should contain the gene IDs.</p></dd>


<dt id="arg-term-name">TERM2NAME<a class="anchor" aria-label="anchor" href="#arg-term-name"></a></dt>
<dd><p>A data frame specifying the term-name mapping for a custom database. The first column should contain the term IDs, and the second column should contain the corresponding term names.</p></dd>


<dt id="arg-mingssize">minGSSize<a class="anchor" aria-label="anchor" href="#arg-mingssize"></a></dt>
<dd><p>A numeric value specifying the minimum size of a gene set to be considered in the enrichment analysis.</p></dd>


<dt id="arg-maxgssize">maxGSSize<a class="anchor" aria-label="anchor" href="#arg-maxgssize"></a></dt>
<dd><p>A numeric value specifying the maximum size of a gene set to be considered in the enrichment analysis.</p></dd>


<dt id="arg-unlimited-db">unlimited_db<a class="anchor" aria-label="anchor" href="#arg-unlimited-db"></a></dt>
<dd><p>A character vector specifying the names of databases that do not have size restrictions.</p></dd>


<dt id="arg-go-simplify">GO_simplify<a class="anchor" aria-label="anchor" href="#arg-go-simplify"></a></dt>
<dd><p>A logical value indicating whether to simplify the GO terms. If <code>TRUE</code>, additional results with simplified GO terms will be returned.</p></dd>


<dt id="arg-go-simplify-cutoff">GO_simplify_cutoff<a class="anchor" aria-label="anchor" href="#arg-go-simplify-cutoff"></a></dt>
<dd><p>A character vector specifying the filter condition for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt id="arg-simplify-method">simplify_method<a class="anchor" aria-label="anchor" href="#arg-simplify-method"></a></dt>
<dd><p>A character vector specifying the method to be used for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt id="arg-simplify-similaritycutoff">simplify_similarityCutoff<a class="anchor" aria-label="anchor" href="#arg-simplify-similaritycutoff"></a></dt>
<dd><p>A numeric value specifying the similarity cutoff for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt id="arg-bpparam">BPPARAM<a class="anchor" aria-label="anchor" href="#arg-bpparam"></a></dt>
<dd><p>A BiocParallelParam object specifying the parallel back-end to be used for parallel computation. Defaults to BiocParallel::bpparam().</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>The random seed for reproducibility. Defaults to 11.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>If input is a Seurat object, returns the modified Seurat object with the enrichment result stored in the tools slot.</p>
<p>If input is a geneID vector with or without geneID_groups, return the enrichment result directly.</p>
<p>Enrichment result is a list with the following component:</p><ul><li><p><code>enrichment</code>: A data.frame containing all enrichment results.</p></li>
<li><p><code>results</code>: A list of <code>gseaResult</code> objects from the DOSE package.</p></li>
<li><p><code>geneMap</code>: A data.frame containing the ID mapping table for input gene IDs.</p></li>
<li><p><code>input</code>: A data.frame containing the input gene IDs and gene ID groups.</p></li>
<li><p><code>DE_threshold</code>: A specific threshold for differential expression analysis (only returned if input is a Seurat object).</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="PrepareDB.html">PrepareDB</a></code> <code><a href="ListDB.html">ListDB</a></code> <code><a href="GSEAPlot.html">GSEAPlot</a></code> <code><a href="RunEnrichment.html">RunEnrichment</a></code> <code><a href="EnrichmentPlot.html">EnrichmentPlot</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreas_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="RunDEtest.html">RunDEtest</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, only.pos <span class="op">=</span> <span class="cn">FALSE</span>, fc.threshold <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The following arguments are not used: drop</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in as.vector(data):</span> no method for coercing this S4 class to a vector</span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunGSEA</span><span class="op">(</span><span class="va">pancreas_sub</span>,</span></span>
<span class="r-in"><span>  group_by <span class="op">=</span> <span class="st">"CellType"</span>, DE_threshold <span class="op">=</span> <span class="st">"p_val_adj &lt; 0.05"</span>,</span></span>
<span class="r-in"><span>  scoreType <span class="op">=</span> <span class="st">"std"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2025-09-08 16:14:03.432714] Start GSEA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in RunGSEA(pancreas_sub, group_by = "CellType", DE_threshold = "p_val_adj &lt; 0.05",     scoreType = "std", db = "GO_BP", species = "Mus_musculus"):</span> Cannot find the DEtest result for the group 'CellType'. You may perform RunDEtest first.</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in GSEAPlot(pancreas_sub, db = "GO_BP", group_by = "CellType", plot_type = "comparison"):</span> No enrichment result found. You may perform RunGSEA first.</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, group_use <span class="op">=</span> <span class="st">"Ductal"</span>, id_use <span class="op">=</span> <span class="st">"GO:0006412"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in GSEAPlot(pancreas_sub, db = "GO_BP", group_by = "CellType", group_use = "Ductal",     id_use = "GO:0006412"):</span> No enrichment result found. You may perform RunGSEA first.</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, group_use <span class="op">=</span> <span class="st">"Endocrine"</span>, id_use <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:0046903"</span>, <span class="st">"GO:0015031"</span>, <span class="st">"GO:0007600"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in GSEAPlot(pancreas_sub, db = "GO_BP", group_by = "CellType", group_use = "Endocrine",     id_use = c("GO:0046903", "GO:0015031", "GO:0007600")):</span> No enrichment result found. You may perform RunGSEA first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Remove redundant GO terms</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunGSEA</span><span class="op">(</span>srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, GO_simplify <span class="op">=</span> <span class="cn">TRUE</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2025-09-08 16:14:03.435693] Start GSEA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in RunGSEA(srt = pancreas_sub, group_by = "CellType", db = "GO_BP",     GO_simplify = TRUE, species = "Mus_musculus"):</span> Cannot find the DEtest result for the group 'CellType'. You may perform RunDEtest first.</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP_sim"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in GSEAPlot(pancreas_sub, db = "GO_BP_sim", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunGSEA first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use a combined database</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunGSEA</span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>,</span></span>
<span class="r-in"><span>  db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEGG"</span>, <span class="st">"WikiPathway"</span>, <span class="st">"Reactome"</span>, <span class="st">"PFAM"</span>, <span class="st">"MP"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  db_combine <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  species <span class="op">=</span> <span class="st">"Mus_musculus"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2025-09-08 16:14:03.43784] Start GSEA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in RunGSEA(srt = pancreas_sub, group_by = "CellType", db = c("KEGG",     "WikiPathway", "Reactome", "PFAM", "MP"), db_combine = TRUE,     species = "Mus_musculus"):</span> Cannot find the DEtest result for the group 'CellType'. You may perform RunDEtest first.</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"Combined"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in GSEAPlot(pancreas_sub, db = "Combined", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunGSEA first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Or use "geneID", "geneScore" and "geneID_groups" as input to run GSEA</span></span></span>
<span class="r-in"><span><span class="va">de_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_CellType</span><span class="op">$</span><span class="va">AllMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in UseMethod("filter"):</span> no applicable method for 'filter' applied to an object of class "NULL"</span>
<span class="r-in"><span><span class="va">gsea_out</span> <span class="op">&lt;-</span> <span class="fu">RunGSEA</span><span class="op">(</span>geneID <span class="op">=</span> <span class="va">de_df</span><span class="op">[[</span><span class="st">"gene"</span><span class="op">]</span><span class="op">]</span>, geneScore <span class="op">=</span> <span class="va">de_df</span><span class="op">[[</span><span class="st">"avg_log2FC"</span><span class="op">]</span><span class="op">]</span>, geneID_groups <span class="op">=</span> <span class="va">de_df</span><span class="op">[[</span><span class="st">"group1"</span><span class="op">]</span><span class="op">]</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2025-09-08 16:14:03.440295] Start GSEA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'de_df' not found</span>
<span class="r-in"><span><span class="fu"><a href="GSEAPlot.html">GSEAPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">gsea_out</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'gsea_out' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

